src/sentry/models/grouphash.py

```diff
@@ -35,3 +42,10 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_grouphash"
         unique_together = (("project", "hash"),)
+
+    @property
+    def metadata(self) -> GroupHashMetadata | None:
+        try:
+            return self._metadata
+        except Exception:
```

Comment:
is there a more specific `Exception` we can use?

Reply:
Yeah, we should use `RelatedObjectDoesNotExist` here

Reply:
(Wrote this before Dan posted)

The type of the error thrown is `sentry.models.grouphash.GroupHash._metadata.RelatedObjectDoesNotExist`, which felt like a real mouthful (and `RelatedObjectDoesNotExist` doesn't seem to be importable on its own, as it's a [dynamically-created type](https://github.com/django/django/blob/a57596e443ecb67140e1a9fc0f0e87446b2d0174/django/db/models/fields/related_descriptors.py#L135-L150)), which is why I went with `Exception`. But it turns out `RelatedObjectDoesNotExist` errors are just `AttributeErrors` (which makes a lot of sense), so I can catch that instead.